app/service/features_report_service.rb

```diff
@@ -0,0 +1,45 @@
+class FeaturesReportService
+  def report
+    {
+      total_live_forms: live_forms.count,
+      live_forms_with_answer_type:,
+      live_pages_with_answer_type:,
+      live_forms_with_payment:,
+      live_forms_with_routing:,
+    }
+  end
+
+private
+
+  def live_forms
+    @live_forms ||= Form.all.filter(&:has_live_version)
+  end
+
+  def pages_on_live_forms
+    @pages_on_live_forms ||= live_forms.flat_map(&:pages)
```

Comment:
I wonder how performant this is with loading in all the results and then filtering through them, and whether we'll run into issues holding a large number of forms/pages in memory. I think it's probably better to let the database do a lot of the heavy lifting.

The most performant way to do it would probably be to have some fairly complex SQL queries to extract a lot of the results in one go. But if we don't want to do that, maybe it's better to run a query per result using ActiveRecord query syntax. And then in future if that becomes too slow, we can consider rewriting as raw SQL?

Reply:
Wasn't sure how well this would perform on a database with more forms, so good to get your view on this. Thanks for the suggestions, think I've got working ActiveRecord queries for all of them now